### PR TITLE
[ENH] Implement Tselect channel selection algorithm

### LINKED ISSUE #3389: [ENH] Implement TSelect as a BaseChannelSelector for multivariate time series
### Describe the feature or idea you want to propose

TSelect 
https://link.springer.com/article/10.1007/s10618-025-01132-4
is a channel selection algorithm that is classifier-agnostic. It would be great to have it in aeon.

Algorithm outline:
**1. Per-channel relevance scoring**
For each channel independently:

extract 6 summary features: min, max, mean, variance, skewness, kurtosis
split the training set into:
learning set: max(100, n_cases // 4)
validation set: remainder
train a simple classifier on the learning set
compute validation ROCAUC

Store:
AUC score per channel
validation predicted probabilities per channel

**2. Irrelevant channel removal**

remove channels with ROCAUC < 0.5
from the remaining channels, remove the worst alpha%

**3. Redundant channel removal**

compare remaining channels using the predictions from step 1
for each class:
rank examples by predicted probability
compute pairwise Spearman rank correlation
average across classes
cluster channels using the threshold corr_threshold
keep the highest-ROCAUC channel from each cluster

Final output:

selected channel indices in channels_selected_

### Describe your proposed solution

TSelect should inherit from BaseChannelSelector and output a subset of the original channels.

```python
class TSelect(BaseChannelSelector):
    def __init__(
        self,
        alpha=0.4,
        corr_threshold=0.7,
        estimator=None,
        random_state=None,
    ):
```

Suggested parameters:

alpha : float, default=0.4
Proportion of the remaining channels to remove after the relevance stage.
corr_threshold : float, default=0.7
Correlation threshold used in redundancy pruning.
estimator : object or None, default=None
Simple per-channel estimator used in the relevance stage. If None, use LogisticRegression.
random_state : int or None, default=None
Random seed for reproducibility.

Possible learned attributes:

channels_selected_
channel_scores_
channel_correlations_
clusters_

binary and multiclass classification support
use sklearn.linear_model.LogisticRegression by default
use roc_auc_score
use a simple threshold-based clustering / greedy grouping first
refine later if needed to better match the paper’s exact clustering lo

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### Reference Issues/PRs

resolves #3389

Implements the TSelect channel selection algorithm in the aeon channel_selection package. Algorithm  is described in
    .. [1] Nuyts, L., Perini, L. and Davis, J. "TSelect: selecting relevant and
        non-redundant channels for multivariate time series classification."
        Data Mining and Knowledge Discovery, 39, 76, 2025.
        https://doi.org/10.1007/s10618-025-01132-4

based on the implementation here
https://github.com/ML-KULeuven/TSelect

### CODE REVIEW

**Comment by TonyBagnall** [comment]
based on authors implementation here 

https://github.com/ML-KULeuven/TSelect

**Comment by TonyBagnall** [comment]
first equivalence test passes 

```python
"""Minimal local comparison of aeon TSelect against ML-KULeuven/TSelect."""

from pathlib import Path
import sys
import warnings

import numpy as np
import pandas as pd


PROJECT_ROOT = Path(__file__).resolve().parents[1]
sys.path.insert(0, str(PROJECT_ROOT))

# Change this to wherever you cloned https://github.com/ML-KULeuven/TSelect
OFFICIAL_TSELECT_REPO = PROJECT_ROOT / "external" / "TSelect"
sys.path.insert(0, str(OFFICIAL_TSELECT_REPO / "src"))

from tselect.channel_selectors.tselect import TSelect as OfficialTSelect
from aeon.transformations.collection.channel_selection import TSelect as AeonTSelect


PARAMS = {
    "irrelevant_selector": True,
    "irrelevant_hard_threshold": 0.5,
    "irrelevant_percentage_to_keep": 0.6,
    "redundant_selector": True,
    "redundant_correlation_threshold": 0.7,
    "validation_size": None,
    "random_state": 0,
}


def make_synthetic_problem(seed=0):
    rng = np.random.default_rng(seed)

    n_cases = 160
    n_channels = 8
    n_timepoints = 60

    y = np.repeat([0, 1], n_cases // 2)
    rng.shuffle(y)

    X = rng.normal(0, 1, size=(n_cases, n_channels, n_timepoints))

    t = np.linspace(0, 2 * np.pi, n_timepoints)
    signal_a = np.sin(t)
    signal_b = np.sign(np.sin(2 * t))

    class_1 = y == 1
    class_0 = y == 0

    # Strong relevant channel.
    X[class_1, 0, :] += 1.25 * signal_a
    X[class_0, 0, :] -= 1.25 * signal_a

    # Redundant with channel 0.
    X[:, 1, :] = X[:, 0, :] + rng.normal(
        0,
        0.05,
        size=(n_cases, n_timepoints),
    )

    # Another relevant channel.
    X[class_1, 2, :] += 0.9 * signal_b
    X[class_0, 2, :] -= 0.9 * signal_b

    # Weak relevant channel.
    X[class_1, 3, 20:40] += 0.35
    X[class_0, 3, 20:40] -= 0.35

    return X, y


def numpy3d_to_official_dataframe(X):
    """Convert aeon NumPy3D to official TSelect pandas MultiIndex format."""
    n_cases, n_channels, n_timepoints = X.shape

    values = np.transpose(X, (0, 2, 1)).reshape(
        n_cases * n_timepoints,
        n_channels,
    )

    index = pd.MultiIndex.from_product(
        [range(n_cases), range(n_timepoints)],
        names=["case", "time"],
    )

    return pd.DataFrame(values, index=index, columns=list(range(n_channels)))


def auc_vector_official(model, n_channels):
    scores = np.full(n_channels, np.nan)

    for channel, auc in model.auc_col.items():
        scores[int(channel)] = float(auc)

    return scores


def auc_vector_aeon(model):
    return np.asarray(model.channel_scores_, dtype=float)


def selected_official(model):
    return sorted(int(channel) for channel in model.selected_channels)


def selected_aeon(model):
    return sorted(int(channel) for channel in model.channels_selected_)


def clusters_official(model):
    return sorted(
        [sorted(int(channel) for channel in cluster) for cluster in model.clusters],
        key=lambda x: (len(x), x),
    )


def clusters_aeon(model):
    return sorted(
        [sorted(int(channel) for channel in cluster) for cluster in model.clusters_],
        key=lambda x: (len(x), x),
    )


def main():
    X, y = make_synthetic_problem(seed=0)
    X_df = numpy3d_to_official_dataframe(X)

    with warnings.catch_warnings():
        warnings.simplefilter("ignore")

        official = OfficialTSelect(**PARAMS)
        official.fit(X_df, pd.Series(y))

        aeon = AeonTSelect(**PARAMS)
        aeon.fit(X, y)

    official_selected = selected_official(official)
    aeon_selected = selected_aeon(aeon)

    official_clusters = clusters_official(official)
    aeon_clusters = clusters_aeon(aeon)

    official_auc = auc_vector_official(official, X.shape[1])
    aeon_auc = auc_vector_aeon(aeon)

    print("Official selected:", official_selected)
    print("aeon selected:    ", aeon_selected)
    print("Selected equal:   ", official_selected == aeon_selected)

    print()
    print("Official clusters:", official_clusters)
    print("aeon clusters:    ", aeon_clusters)
    print("Clusters equal:   ", official_clusters == aeon_clusters)

    print()
    print("Official AUC:", np.round(official_auc, 6))
    print("aeon AUC:    ", np.round(aeon_auc, 6))

    mask = ~np.isnan(official_auc) & ~np.isnan(aeon_auc)
    if np.any(mask):
        print()
        print("Max AUC abs diff: ", np.max(np.abs(official_auc[mask] - aeon_auc[mask])))
        print("Mean AUC abs diff:", np.mean(np.abs(official_auc[mask] - aeon_auc[mask])))
```
Official selected: [3, 5, 7]
  aeon selected:     [3, 5, 7]
  Selected equal:    True
  Clusters equal:    True
  Max AUC abs diff:  0.011123470522803158
  Mean AUC abs diff: 0.006674082313681902

**Comment by TonyBagnall** [comment]
run time equivalent, original uses cython

image

**Comment by TonyBagnall** [comment]
@LorenNuyts comments welcome

**Comment by TonyBagnall** [comment]
sanity check equivalence 

image

**Comment by TonyBagnall** [comment]
image

**Review by MatthewMiddlehurst — APPROVED** [review]
LGTM. Thank you for the comparisons to the original.

One optional addition: mention the license when attributing the original repo if code is used from it
https://github.com/ML-KULeuven/TSelect/blob/main/LICENSE

**Comment by LorenNuyts** [comment]
This already looks good. A few notes on the results:

**AUC differences (~0.011 max)** are expected and reasonable. The aeon implementation presumably reimplements
the six summary statistics independently rather than calling TSFuse's `SinglePassStatistics`
directly. Small numerical differences from that won't affect correctness as long as the relative
ordering between channels is preserved, which the matching selected channels and clusters confirm.

**Runtime difference for >60 channels**. The official implementation uses a Cython-accelerated
pairwise Spearman rank correlation (in `rank_correlation/rank_correlation.py`). For large channel
counts the O(n²) pairwise correlation step dominates, so the gap is expected.
